### PR TITLE
[ Feat ] Arcjet Security Implementation

### PR DESCRIPTION
**Pull request by dev-xo**
This PR implements Arcjet Security into Remix SaaS via `remix-init`.  

Remix SaaS should be initialized with `npx create-remix-saas@latest`, a command that will take care of a few things, one of those allowing the user to opt in for Arcjet Security, replacing the required files to properly implement and initialize Arcjet into Remix SaaS.  

Feel free to provide new documentation or anything else that could add value to the whole feature/implementation @davidmytton.  

The following video showcases how the implementation would look for the end user.

https://github.com/user-attachments/assets/18e95c68-5aa0-4396-b980-21dc68edbe45
- - -
With that said, I will also recover the contact of a few people who decided to sponsor Remix SaaS before. Sadly, my paperwork wasn't in place (personal documentation), and some of those opportunities were simply skipped by me.  

In the same way, Fly.io will publish a blog post sooner rather than later, as far as I know, showcasing Remix SaaS as part of some Fly.io apps.  

Let me know anything else you may want to talk about, and feel free to provide new documentation, links for the SVG, or anything else.

### CODE REVIEW

File: remix.init/index.js [hunk 1]
```diff
@@ -65,20 +67,101 @@ function getPackageManagerCommand(packageManager) {
...
+  )
+}
+
+/**
+ * Prompts user for Arcjet setup preferences.
+ */
+async function promptUser() {
+  try {
+    const optInArcjet = await confirm({
+      message: chalk.blue('Would you like to use Arcjet Security?'),
```
Comment (davidmytton) [inline remix.init/index.js:97]:
```suggestion
      message: chalk.blue('Would you like to use Arcjet?\n\nAdding Arcjet provides bot protection, spam signup detection and attack prevention for out of the box for free. Other features e.g. rate limiting & PII redaction can be enabled.'),
```

File: remix.init/index.js [hunk 2]
```diff
@@ -96,53 +179,138 @@ async function initEnvFile(rootDirectory) {
     `HONEYPOT_ENCRYPTION_SEED="${generateRandomHexadecimalString(64)}"`,
   )
 
+  if (useArcjet) {
+    newEnv += `\n\n# Arcjet Security (Optional) - Rate limiting, bot detection, and request filtering.
```
Comment (davidmytton) [inline remix.init/index.js:183]:
```suggestion
    newEnv += `\n\n# Arcjet Security (Optional) - Bot detection, spam signup detection, and attack prevention.
```

**Comment by davidmytton** [comment]
Looks good. I've suggested some changes to the wording.

The docs from https://github.com/dev-xo/remix-saas/pull/362/files#diff-2abc6729452120f62d2f8d7dceabdaac18819d3346b66a3acc89c1ef894922ff can be moved over as well - it only needs minor changes to reflect that this needs to be enabled and how to do it manually.

You can use these SVG images:

```html
<a href="https://arcjet.com" target="_arcjet-home">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://arcjet.com/logo/arcjet-dark-lockup-voyage-horizontal.svg">
    <img src="https://arcjet.com/logo/arcjet-light-lockup-voyage-horizontal.svg" alt="Arcjet Logo" height="128" width="auto">
  </picture>
</a>
```

**Comment by dev-xo** [comment]
All right, will finish this tomorrow. We got the original docs, added a few images of Arcjet on the main docs, and simply moved what was on the original PR here, with the addition of the updated `remix-init`.

I want to test this on a private repository (initializing the template and checking that everything is in place), and with that, we may be able to merge into `main`.

Also, I want to create some docs that could showcase a manual initialization of Arcjet, in case some folks simply want to opt in later on, either by noticing they can do so based on the docs provided or simply because they discovered Arcjet.  
- - -  
Again, this looks good to me, especially considering that Remix SaaS has been created to be a community project and one that I know some folks appreciate, and maintaining that is a way to pay back some of the support they provided.

Will also reach back to some folks who wanted to support the project before, in case they are still up for it, and that will simply do it for now.

There are plans to move Remix SaaS to Cloudflare, React-Router 7, and simply maintain and improve the project over time, allowing it to be discovered by some new folks, which is great, and if that also helps Arcjet in some way, even better.

With that said, will let you know to give the initialization a test, and with that done, the whole feature would be ready to be pushed.

File: README.md [hunk 3]
```diff
@@ -42,3 +42,14 @@ If you found **Remix SaaS** helpful, consider supporting it with a ⭐ [Star](ht
 ## Acknowledgments
 
 Special thanks to [@mw10013](https://github.com/mw10013) who has been part of the Remix SaaS development.
+
+## Sponsors
+
+Remix SaaS is proudly supported by [Arcjet](https://arcjet.com/).
```
Comment (davidmytton) [inline README.md:48]:
```suggestion
Remix SaaS is proudly supported by [Arcjet](https://launch.arcjet.com/hdXzPbO).
```

File: README.md [hunk 4]
```diff
@@ -42,3 +42,14 @@ If you found **Remix SaaS** helpful, consider supporting it with a ⭐ [Star](ht
...
+
+<a href="https://arcjet.com" target="_arcjet-home">
```
Comment (davidmytton) [inline README.md:50]:
```suggestion
<a href="https://launch.arcjet.com/hdXzPbO" target="_arcjet-home">
```

File: docs/README.md [hunk 5]
```diff
@@ -142,3 +142,14 @@ Deployment and some other docs are available in the [Deployment](./guide/09-depl
 # Support
 
 If you found **Remix SaaS** helpful, consider supporting it with a ⭐ [Star](https://github.com/dev-xo/remix-saas). It helps the repository grow and provides the required motivation to continue maintaining the project. Thank you!
+
+## Sponsors
+
+Remix SaaS is proudly supported by [Arcjet](https://arcjet.com/).
+
+<a href="https://arcjet.com" target="_arcjet-home">
```
Comment (davidmytton) [inline docs/README.md:150]:
```suggestion
<a href="https://launch.arcjet.com/hdXzPbO" target="_arcjet-home">
```

File: docs/guide/05-utilities.md [hunk 6]
```diff
@@ -34,6 +34,29 @@ You can authenticate as `admin` by using the following credentials:
 
 The default admin email can be changed in the `/prisma/seed.ts` file.
 
+## Arcjet Security
+
+Remix SaaS includes an optional integration with [Arcjet](https://arcjet.com/), a security-as-code product that helps secure your application with bot protection, rate limiting, and signup form spam protection.
```
Comment (davidmytton) [inline docs/guide/05-utilities.md:39]:
```suggestion
Remix SaaS includes an optional integration with [Arcjet](https://launch.arcjet.com/hdXzPbO), a security-as-code product that helps secure your application with bot protection, rate limiting, and signup form spam protection.
```

File: remix.init/arcjet/README.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
+<a href="https://arcjet.com" target="_arcjet-home">
```
Comment (davidmytton) [inline remix.init/arcjet/README.md:1]:
```suggestion
<a href="https://launch.arcjet.com/hdXzPbO" target="_arcjet-home">
```

File: remix.init/arcjet/README.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+[Arcjet][arcjet] helps developers protect their apps in just a few lines
+of code. Bot detection. Rate limiting. Email validation. Attack protection.
+Data redaction. A developer-first approach to security.
+
+## Getting started
+
+1. Sign up for a free account at [arcjet.com][arcjet].
+2. [Follow the quick start][quick-start] for your platform.
+
+[arcjet]: https://arcjet.com
```
Comment (davidmytton) [inline remix.init/arcjet/README.md:19]:
```suggestion
[arcjet]: https://launch.arcjet.com/hdXzPbO
```

File: remix.init/index.js [hunk 9]
```diff
@@ -101,48 +186,127 @@ async function initEnvFile(rootDirectory) {
...
+    description: isDarkTheme ? chalk.gray : chalk.dim.black,
+  }
+
+  console.log('\n')
+
+  if (optInArcjet) {
+    console.log(colors.header('⚡ Next Steps:'))
+    console.log(colors.subHeader('📚 Arcjet Security Setup:'))
+    console.log(
+      `${colors.bullet('1.')} Visit your Dashboard: ${colors.link('https://arcjet.com')}`,
```
Comment (davidmytton) [inline remix.init/index.js:259]:
```suggestion
      `${colors.bullet('1.')} Sign up to Arcjet ${colors.link('https://launch.arcjet.com/hdXzPbO')}`,
```

**Comment by davidmytton** [comment]
How can I test the init script? I tried `node index.js` from within `remix.init`, but it throws an error due to the commonjs module syntax.

**Comment by dev-xo** [comment]
Thanks for the updates on the docs (redirect to the arcjet.com ref).

About testing it, not sure if you have tried it on this repo; currently, there is no available way to test it here, and that's why the `remix-init-arcjet` repository was created.

Remix runs `remix-init` after template initialization, allowing us to customize it, as we have done in this case. I'm unsure of how Remix runs the `remix-init/index.js` script, although probably should be similar to what we do in `remix-init-arcjet`.

In order to properly test it, I will set up another repo with the updated changes on this PR, and by simply running `npx create-remix-saas-dev@latest`, we should be able to successfully initialize the repository, opting in for Arcjet.

Will have that done by tomorrow, allowing you to give it a test and also share some feedback, in order to merge into `main` after ensuring everything works as expected.

**Comment by dev-xo** [comment]
This is a bit picky, @davidmytton, but can the ref code be simply `arcjet.com/?ref=remix-saas`? Again, just a little preference, but I think the URL could look a bit cleaner, and that could have a little impact in the long run.

If so, feel free to simply share the new ref here, and I will update those wherever required in the docs.

**Comment by dev-xo** [comment]
All right, this should be working.  
- You can give the `dev` repository a try with: `npx create-remix-saas-dev@latest`.  

I want to polish the Arcjet implementation a bit, update the comments provided, and simply check the implementation to learn more about it. Overall, the basics and the opt-in should already be working as expected.  

Let me know how it goes.
https://github.com/user-attachments/assets/dc0ef756-8279-47a2-8eac-ac4c1ce137db

**Comment by davidmytton** [comment]
Looks good - all works on my side.

Given that users are opting in to Arcjet, should we change the code to throw if the `ARCJET_KEY` environment variable isn't set like it does with Stripe?

**Comment by dev-xo** [comment]
Yes, the moment the users have opted for ARCJET, the key may be required, good point. Gonna clean up the comments a bit and check on the base implementation (the files we are replacing), and with that done, we may be ready to merge into `main`.

It would be great to also have a step-by-step guide in the docs on how to implement ARCJET for those who opted to do it later, as that could be really helpful.

**Comment by dev-xo** [comment]
All right, this should be done @davidmytton.

Feel free to give it a last install to the `dev` repository, although the implementation should be the same, just a bit polished.

- Also, as mentioned before, if you want to provide a new ref `remix-saas` in order to have a cleaner URL, I can update that across the docs.

The integration ended up pretty good, the default to opt-in Arcjet is `true`, so I would say that most, if not all, the installs may include Arcjet, which is good.

Let me know if you want to test anything else; otherwise, this may be ready to be merged into the `main` branch.

**Comment by davidmytton** [comment]
Thanks, all looks good. Let's keep the current URL for now as it tracks clicks better.

Once this is merged, I'll open a new PR with docs improvements for you to review.

**Comment by dev-xo** [comment]
All right, feel free to reach back with that docs @davidmytton, and we can even create a `10-arcjet-security.md` specific one, if you prefer it, with the configuration, the step-by-step guide, or anything else related to Arcjet.

Will wait for the Sponsor then, and we can merge into `main`, as everything should be ready.

**Comment by dev-xo** [comment]
Thanks to the sponsor @davidmytton! I think the implementation ended up well for both, especially considering that most, if not all, users will opt for Arcjet, as either by pressing `enter` or `Y`, the user is in.

Merging this into `main`. Please feel free to reach out with the updated docs or anything else that may fit the implementation.

**Comment by dev-xo** [comment]
Remix Discord Announcement: [Discord Post](https://discord.com/channels/770287896669978684/1048267992137466006/1319709829186715759)

**Pull request merged**